src/System.IO.FileSystem/src/System/IO/Win32FileSystemEnumerable.cs

```diff
@@ -492,13 +496,13 @@ private static String GetNormalizedSearchCriteria(String fullSearchString, Strin
             return searchCriteria;
         }
 
-        private static String GetFullSearchString(String fullPath, String searchPattern)
+        private static string GetFullSearchString(string fullPath, string searchPattern)
         {
             Contract.Requires(fullPath != null);
             Contract.Requires(searchPattern != null);
 
             PathHelpers.ThrowIfEmptyOrRootedPath(searchPattern);
-            String tempStr = Path.Combine(fullPath, searchPattern);
+            string tempStr = Path.Combine(fullPath, searchPattern);
```

Comment:
Since searchPattern isn't null or empty, Path.Combine isn't going to effect the end of searchPattern, right?  Meaning whether tempStr ends in a directory / volume separator is the same as whether searchPattern ends in a directory / volume separator?  Assuming yes, we can avoid another string allocation here in the case where we need to append the *, e.g. something like:
```C#
private static string GetFullSearchString(string fullPath, string searchPattern)
{
    ...
    PathHelpers.ThrowIfEmptyOrRootedPath(searchPattern);
    char lastChar = searchPattern[searchPattern.Length - 1];
    return PathHelpers.IsDirectorySeparator(lastChar) || lastChar == Path.VolumeSeparatorChar ?
        Path.Combine(fullPath, searchPattern, "*") :
        Path.Combine(fullPath, searchPattern);
}
```
Are there any cases with that I'm not thinking about?